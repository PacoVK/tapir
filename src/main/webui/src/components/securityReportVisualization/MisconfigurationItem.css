.severity-LOW {
  background-color: #5fbb31 !important;
}
.severity-MEDIUM {
  background-color: #e9c600 !important;
}
.severity-HIGH {
  background-color: #ff8800 !important;
}
.severity-CRITICAL {
  background-color: #e40000 !important;
}
.severity-UNKNOWN {
  background-color: #747474 !important;
}
