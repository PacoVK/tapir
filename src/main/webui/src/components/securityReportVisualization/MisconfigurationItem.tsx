import React from "react";
import {
  Accordion,
  AccordionDetails,
  AccordionSummary,
  Link,
  Typography,
} from "@mui/material";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import { Misconfiguration } from "../../types";
import "./MisconfigurationItem.css";

const MisconfigurationItem = ({
  misconfiguration,
  keyIdentifier,
}: {
  misconfiguration: Misconfiguration;
  keyIdentifier: string;
}) => {
  return (
    <Accordion key={misconfiguration.resource + keyIdentifier}>
      <AccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls={`${
          misconfiguration.ruleDescription + keyIdentifier
        }-content`}
        id={`${misconfiguration.ruleDescription + keyIdentifier}-header`}
        className={`severity-${misconfiguration.severity}`}
      >
        <Typography>
          {misconfiguration.resource}, Line:{" "}
          {misconfiguration.location.startLine} -{" "}
          {misconfiguration.location.endLine}
        </Typography>
      </AccordionSummary>
      <AccordionDetails>
        <Typography>
          <b>Severity:</b> {misconfiguration.severity}
        </Typography>
        <Typography>
          <b>Violation:</b> {misconfiguration.ruleDescription}
        </Typography>
        <Typography>
          <b>Impact:</b> {misconfiguration.impact}
        </Typography>
        <Typography>
          <b>Solution:</b> {misconfiguration.resolution}
        </Typography>
        <Typography>
          <b>Resource:</b> {misconfiguration.resource}
        </Typography>
        <Typography>
          <b>From line:</b> {misconfiguration.location.startLine}
        </Typography>
        <Typography>
          <b>To line:</b> {misconfiguration.location.endLine}
        </Typography>
        <Link
          href={misconfiguration.links[misconfiguration.links.length - 1]}
          rel="noopener"
          target="_blank"
        >
          Read more
        </Link>
      </AccordionDetails>
    </Accordion>
  );
};

export default MisconfigurationItem;
