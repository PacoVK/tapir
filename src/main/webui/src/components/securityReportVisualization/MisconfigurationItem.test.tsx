import { render } from "@testing-library/react";
import MisconfigurationItem from "./MisconfigurationItem";

describe("<MisconfigurationItem /> spec", () => {
  it("renders the MisconfigurationItem", () => {
    const container = render(
      <MisconfigurationItem
        misconfiguration={{
          severity: "CRITICAL",
          ruleDescription: "foo",
          links: "https://fake.com",
          resolution: "bar",
          resource: "aws.securitygroup",
          impact: "baz",
          location: {
            startLine: 38,
            endLine: 40,
            filename: "any.tf",
          },
          description: "description",
        }}
        keyIdentifier={"foo-bar-baz"}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
