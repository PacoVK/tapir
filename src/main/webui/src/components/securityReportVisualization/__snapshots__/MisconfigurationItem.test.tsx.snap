// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MisconfigurationItem /> spec renders the MisconfigurationItem 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-controls="foofoo-bar-baz-content"
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters severity-CRITICAL css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
          id="foofoo-bar-baz-header"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              aws.securitygroup
              , Line:
               
              38
               -
               
              40
            </p>
          </div>
          <div
            class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ExpandMoreIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                aria-labelledby="foofoo-bar-baz-header"
                class="MuiAccordion-region"
                id="foofoo-bar-baz-content"
                role="region"
              >
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      Severity:
                    </b>
                     
                    CRITICAL
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      Violation:
                    </b>
                     
                    foo
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      Impact:
                    </b>
                     
                    baz
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      Solution:
                    </b>
                     
                    bar
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      Resource:
                    </b>
                     
                    aws.securitygroup
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      From line:
                    </b>
                     
                    38
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                  >
                    <b>
                      To line:
                    </b>
                     
                    40
                  </p>
                  <a
                    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                    href="m"
                    rel="noopener"
                    target="_blank"
                  >
                    Read more
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
    >
      <div
        aria-controls="foofoo-bar-baz-content"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters severity-CRITICAL css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
        id="foofoo-bar-baz-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            aws.securitygroup
            , Line:
             
            38
             -
             
            40
          </p>
        </div>
        <div
          class="MuiAccordionSummary-expandIconWrapper css-yw020d-MuiAccordionSummary-expandIconWrapper"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ExpandMoreIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </div>
      </div>
      <div
        class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="foofoo-bar-baz-header"
              class="MuiAccordion-region"
              id="foofoo-bar-baz-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    Severity:
                  </b>
                   
                  CRITICAL
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    Violation:
                  </b>
                   
                  foo
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    Impact:
                  </b>
                   
                  baz
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    Solution:
                  </b>
                   
                  bar
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    Resource:
                  </b>
                   
                  aws.securitygroup
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    From line:
                  </b>
                   
                  38
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                >
                  <b>
                    To line:
                  </b>
                   
                  40
                </p>
                <a
                  class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
                  href="m"
                  rel="noopener"
                  target="_blank"
                >
                  Read more
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
